src/client/Microsoft.Identity.Client/AppConfig/CacheOptions.cs

```diff
@@ -50,5 +65,35 @@ public CacheOptions(bool useSharedCache)
         /// </remarks>
         public bool UseSharedCache { get; set; }
 
+        private long? _sizeLimit;
+
+        /// <summary>
+        /// Total token cache size limit in bytes for both app and user token caches.
+        /// </summary>
+        /// <remarks>
+        /// Once the limit is reached, either the app or user cache will be fully cleared, depending on which was most recently used.
+        /// MSAL doesn't calculate the exact memory usage and uses approximations of the token sizes. 
+        /// For instance, app token cache entry is approximately at least 4500 bytes; user access token entry - 6500 bytes,
+        /// user refresh token entry - 3700 bytes.
+        /// IMPORTANT: Monitor app health metrics (including memory usage) and cache performance (<see href="https://aka.ms/msal-net-token-cache-serialization"/>)
+        /// and adjust size limit accordingly.
+        /// </remarks>
+        public long? SizeLimit
+        {
+            get => _sizeLimit;
+            set
+            {
+                ValidateSizeLimit(value);
+                _sizeLimit = value;
+            }
+        }
+
+        private void ValidateSizeLimit(long? sizeLimit)
+        {
+            if (!sizeLimit.HasValue || sizeLimit < 0)
```

Comment:
Why not allow users to set NULL ?

Reply:
Yea, I mean if they don't want to set the limit, use the other constructor. I guess null is technically valid here.